Neos.Neos/Classes/Domain/Service/RenderingModeService.php

```diff
@@ -85,16 +90,14 @@ public function findDefault(): RenderingMode
      */
     public function findByName(string $modeName): RenderingMode
     {
-        if ($modeName === 'frontend') {
-            return RenderingMode::createFrontend();
-        }
-        if (isset($this->editPreviewModes[$modeName])) {
-            return RenderingMode::createFromConfiguration($modeName, $this->editPreviewModes[$modeName]);
-        }
-        throw new Exception(
-            'The requested interface render mode "' . $modeName . '" is not configured.'
+        return $this->instances[$modeName] ??= match (true) {
```

Comment:
I kinda dislike that you put the runtime cache in `instances` into one live with code that populates this.

Needlessly harder to read and unterstand. Offcourse the runtime cache makes sense.

Reply:
suggestion:
```
	public function findByName(string $modeName): RenderingMode
    {
        if ($instance = $this->instances[$modeName]) {
            return $instance;
        }
        if ($modeName === RenderingMode::FRONTEND) {
            $this->instances[$modeName] = RenderingMode::createFrontend();
           } else {
               $this->instances[$modeName] = RenderingMode::createFromConfiguration($modeName, $this->editPreviewModes[$modeName]);
          }
          return $this->instances[$modeName];        
    }
```

Reply:
`??=` and `match($modeName)` would be fine for me. What is imho too much is the match true you have to use to thwrow the exception. Will adjust and the merge this into the other pr. 

Funny is that even the php documentation does not explain that operator other than mentioning its existence.

Reply:
This would be fine for me ... but it will not throw on unconfigured modes
```php 
return $this->instances[$modeName] ??= match ($modeName) {
   RenderingMode::FRONTEND => RenderingMode::createFrontend(),
   default => RenderingMode::createFromConfiguration($modeName, $this->editPreviewModes[$modeName])
};
```